### PR TITLE
Update libsass to v3.6.0 and fixes various issues (windows, cross-compilation, binary gems)

### LINKED ISSUE #18: Windows?
Coincidentally the gem seems to install just fine however when I actually try to use it as a drop-in replacement for sass-rails I get the following upon trying to start a Rails server:

Uncaught exception: Could not open library 'C:/tools/ruby21/lib/ruby/gems/2.1.0/gems/sassc-1.6.0/ext/libsass/lib/libsass.so': The specified module could not be found.

Thoughts?

### LINKED ISSUE #91: 1.12.1 - :not selector issue
We recently upgraded from 1.11.4 to 1.12.1 and noticed a bug that seems to be around the :not selector. 

**When using 1.11.4:**
SCSS
```
.form-group {
  label:not(.col-form-label,.form-check-label,.radio-inline,.form-label-static) {
    position: absolute;
    top: 0.6rem;
    left: 1.1rem;
  }
}
```
Output CSS
```
.form-group label:not(.col-form-label):not(.form-check-label):not(.radio-inline):not(.form-label-static) {
  position: absolute;
  top: 0.6rem;
  left: 1.1rem;
}

```

**When using 1.12.1**
SCSS (Same as above)
Output CSS
```
.form-group label {
  position: absolute;
  top: 0.6rem;
  left: 1.1rem;
}
```

Please let me know if you need any more information, or if maybe this is a symptom of a different problem. Thanks!

EDIT: It looks like this only affects nested :not selectors. When I change the SCSS to 

```
.form-group label:not(.col-form-label,.form-check-label,.radio-inline,.form-label-static) {
    position: absolute;
    top: 0.6rem;
    left: 1.1rem;
}
```
Then it works properly.

### LINKED ISSUE #93: Configuring Sass with multiple valid load paths leads to SyntaxError on Windows
Hello Guys :wave:

## Background
This issue was discovered while [migrating Jekyll's built-in sass-converter](https://github.com/jekyll/jekyll-sass-converter/pull/75/files) to use `sassc-2.0.0`.
Jekyll sites typically have two sass-load-paths pre-configured:
  - `_sass` directory at the site's source root.
  - `_sass` directory at the site's theme root (if the site uses a theme-gem e.g. [Minima](https://rubygems.org/gems/minima))

Out-of-the box, a Jekyll site using the `sassc` branch (the migration PR linked above) yields no errors on both Unix based systems and Windows because the site is only loading sass files from the theme-gem.
The issue arises when the Windows user adds a `_sass` directory at their site's source root (in order to override styles defined by the theme-gem).

Now there exists two valid entries, in the `load_paths` array,

## Summary
SassC on Windows attempts to import a sass partial from one of the load_paths and just `raises` instead of checking the other load path for the imported partial as well.

## Misc.
- I've a temporary test repo at https://github.com/ashmaroli/sassc-test, wired up with both Travis CI and AppVeyor CI to compare logs across the two platforms.
- This issue is currently preventing the Jekyll Team from migrating to sassc-ruby, and would love your assistance in resolving the issue.
Thanks.

/cc @am11

### LINKED ISSUE #95: libsass.so stays in ext/ - but should be in lib/
I would have provided a PR if I knew rake a little better. But openSUSE packages gems so that ext/ is left as sources not for production.

Every other gem moves/copies libraries required to lib/, sassc sticks out there.

### LINKED ISSUE #105: Dropping the rake runtime dependency
Currently, this gem depends on `rake`:
https://github.com/sass/sassc-ruby/blob/bc66f2a4af15d6e89f93a919bcc23cce362c945a/sassc.gemspec#L34

Rake is typically a development-only dependency.
The gem actually uses very little of rake at runtime, and the purpose seems to be just to build the C extension.

I think moving the logic from https://github.com/sass/sassc-ruby/blob/bc66f2a4af15d6e89f93a919bcc23cce362c945a/lib/tasks/libsass.rb to `ext/extconf.rb` and removing the couple rake-specific lines should be enough.

### PR DESCRIPTION
**Pull request by glebm**
Updates libsass to v3.6.0

Also:
1. Adds Windows RubyInstaller compilation support via `rake-compiler`. Fixes #18.
2. Fixes `load_paths` separator on Windows. Fixes #93.
3. Changes the location of `libsass.so` from `ext/` to `lib/`. Fixes #95.
4. Easy binary release cross-compilation via `rake-compiler-dock`. Simply install Docker and run `bundle exec rake gem:native`.
5. Removes runtime rake dependency. Fixes #105.

Some important fixes in libsass v3.6.0:

1. The :not selector. Fixes #91. https://github.com/sass/libsass/issues/2697
2. Default precision changed to 10. https://github.com/sass/libsass/issues/2716
3. Now builds with `cc` instead of hard-coding `gcc`. https://github.com/sass/libsass/issues/2707
4. Building on Linux and Solaris. https://github.com/sass/libsass/pull/2720
5. Many other Sass compatibility issues, and dozens of memory leaks and segfaults.

### CODE REVIEW

**Comment by wmoxam** [comment]
Is there anything preventing either this or #94 ?

**Comment by bolandrm** [comment]
Thus far, I've been targeting only libsass releases (rather than specific commits).  It doesn't look like there's been a release containing the commit that this PR points to yet.   I suppose we could merge #94

**Comment by ahorek** [comment]
after this change, I can't compile it on Windows, is it intentional?

**Comment by glebm** [comment]
> after this change, I can't compile it on Windows, is it intentional?

It's not intentional, you should be able to compile on Windows with this PR.
Can you provide more information about the error you're getting?

**Comment by ahorek** [comment]
sure, see gem_make.out
<details>
<summary>master</summary>
<pre>
current directory: c:/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext
c:/Ruby24-x64/bin/ruby.exe -rrubygems c:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/rake-12.3.2/exe/rake RUBYARCHDIR\=c:/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/extensions/x64-mingw32/2.4.0/sassc-ruby-58d09fe58563 RUBYLIBDIR\=c:/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/extensions/x64-mingw32/2.4.0/sassc-ruby-58d09fe58563
cd libsass
git submodule update --init
Submodule 'ext/libsass' (git://github.com/sass/libsass.git) registered for path './'
Cloning into 'C:/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass'...
Submodule path './': checked out '9cfe0df975945325e05d6dd39ee98224898d9a79'
make lib/libsass.so
mkdir lib
gcc -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/cencode.o src/cencode.c
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/ast.o src/ast.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/node.o src/node.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/context.o src/context.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/constants.o src/constants.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/functions.o src/functions.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/color_maps.o src/color_maps.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/environment.o src/environment.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/ast_fwd_decl.o src/ast_fwd_decl.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/bind.o src/bind.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/file.o src/file.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/util.o src/util.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/json.o src/json.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/units.o src/units.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/values.o src/values.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/plugins.o src/plugins.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/position.o src/position.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/lexer.o src/lexer.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/parser.o src/parser.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/prelexer.o src/prelexer.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/eval.o src/eval.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/expand.o src/expand.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/listize.o src/listize.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/cssize.o src/cssize.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/extend.o src/extend.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/output.o src/output.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/inspect.o src/inspect.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/emitter.o src/emitter.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/check_nesting.o src/check_nesting.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/remove_placeholders.o src/remove_placeholders.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/sass.o src/sass.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/sass_util.o src/sass_util.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/sass_values.o src/sass_values.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/sass_context.o src/sass_context.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/sass_functions.o src/sass_functions.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/sass2scss.o src/sass2scss.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/backtrace.o src/backtrace.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/operators.o src/operators.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/to_c.o src/to_c.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/to_value.o src/to_value.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/source_map.o src/source_map.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/subset_map.o src/subset_map.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/error_handling.o src/error_handling.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/memory/SharedPtr.o src/memory/SharedPtr.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/utf8_string.o src/utf8_string.cpp
g++ -Wall -O2 -DLIBSASS_VERSION="\"3.5.2\"" -std=gnu++0x -I /c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-58d09fe58563/ext/libsass/include -c -o src/base64vlq.o src/base64vlq.cpp
g++ -shared -Wall -O2 -Wl,-undefined,error -std=gnu++0x -static -static-libgcc -static-libstdc++ -o lib/libsass.so src/cencode.o src/ast.o src/node.o src/context.o src/constants.o src/functions.o src/color_maps.o src/environment.o src/ast_fwd_decl.o src/bind.o src/file.o src/util.o src/json.o src/units.o src/values.o src/plugins.o src/position.o src/lexer.o src/parser.o src/prelexer.o src/eval.o src/expand.o src/listize.o src/cssize.o src/extend.o src/output.o src/inspect.o src/emitter.o src/check_nesting.o src/remove_placeholders.o src/sass.o src/sass_util.o src/sass_values.o src/sass_context.o src/sass_functions.o src/sass2scss.o src/backtrace.o src/operators.o src/to_c.o src/to_value.o src/source_map.o src/subset_map.o src/error_handling.o src/memory/SharedPtr.o src/utf8_string.o src/base64vlq.o -lm -lstdc++
cd -
</pre>
</details>

<details>
<summary>this branch</summary>
<pre>
current directory: c:/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-b0b9592d6e4c/ext
c:/Ruby24-x64/bin/ruby.exe -I c:/Ruby24-x64/lib/ruby/site_ruby/2.4.0 -r ./siteconf20190505-2224-cjg2c9.rb extconf.rb

current directory: c:/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-b0b9592d6e4c/ext
make "DESTDIR=" clean
make -C 'c:/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-b0b9592d6e4c/ext/libsass' clean
make[1]: Entering directory `/c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-b0b9592d6e4c/ext/libsass'
make[1]: *** No rule to make target `clean'.  Stop.
make[1]: Leaving directory `/c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-b0b9592d6e4c/ext/libsass'
make: *** [clean] Error 2

current directory: c:/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-b0b9592d6e4c/ext
make "DESTDIR="
make -C 'c:/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-b0b9592d6e4c/ext/libsass' lib/libsass.so
make[1]: Entering directory `/c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-b0b9592d6e4c/ext/libsass'
make[1]: *** No rule to make target `lib/libsass.so'.  Stop.
make[1]: Leaving directory `/c/Ruby24-x64/lib/ruby/gems/2.4.0/bundler/gems/sassc-ruby-b0b9592d6e4c/ext/libsass'
make: *** [libsass.so] Error 2

make failed, exit code 2 
</pre>
</details>

**Comment by xzyfer** [comment]
We'll hold off tagging a LibSass release incase upstream changes are
required.

**Comment by glebm** [comment]
Should be fixed now!

File: ext/extconf.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+gem_root = File.expand_path('..', __dir__)
+libsass_dir = File.join(gem_root, 'ext', 'libsass')
+
+if !File.directory?(libsass_dir) || Dir.entries(libsass_dir).empty?
```
[the remark this answers is not in the document]
Comment (glebm) [reply ext/extconf.rb]: Good catch, fixed!

**Comment by glebm** [comment]
@ahorek Should be fixed now. Conflicts also resolved.

**Comment by glebm** [comment]
@ahorek Should be fixed now. Conflicts also resolved.

**Comment by ahorek** [comment]
Thanks @glebm

**Comment by glebm** [comment]
I've added easy cross-platform release compilation via `rake-compiler-dock`.

Simply have Docker installed and run this to compile binary gems:

    bundle exec rake gem:native

**Comment by glebm** [comment]
@bolandrm Perhaps we should merge this and release a `beta` gem?

This will let users try out the new pre-built gems and report any issues before the final release.

**Comment by xzyfer** [comment]
I'll try to tag a new LibSass release this week.

**Comment by betesh** [comment]
@xzyfer Sounds like we're just waiting for a release at this point.  Any update on that?

**Comment by xzyfer** [comment]
Apologies I fell behind. I just released https://github.com/sass/libsass/releases/tag/3.6.0

**Comment by glebm** [comment]
Should be good to merge and release!

Please also merge #117 before releasing!

To compile binary gems, simply run:

```bash
bundle exec rake gem:native
```

This will cross-compile gems for Windows (mingw / RubyInstaller) and Linux using Docker + VirtualBox.

**Comment by bolandrm** [comment]
@glebm i'm a bit confused- are the precompiled binaries going to be included as part of the ruby gem?

**Comment by bolandrm** [comment]
@glebm hitting the following issue on osx:

```
bundle exec rake compile:libsass
cd tmp/x86_64-darwin17/libsass/2.5.1
/usr/local/bin/gmake
gmake: 'libsass.so' is up to date.
cd -
install -c tmp/x86_64-darwin17/libsass/2.5.1/libsass.bundle lib/sassc/libsass.bundle
rake aborted!
Errno::ENOENT: No such file or directory @ rb_file_s_stat - tmp/x86_64-darwin17/libsass/2.5.1/libsass.bundle
/Users/bolandrm/www/gems/sassc-ruby/vendor/bundle/gems/rake-compiler-1.0.7/lib/rake/extensiontask.rb:146:in `block in define_compile_tasks'
/Users/bolandrm/www/gems/sassc-ruby/vendor/bundle/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
/Users/bolandrm/.rbenv/versions/2.5.1/bin/bundle:23:in `load'
/Users/bolandrm/.rbenv/versions/2.5.1/bin/bundle:23:in `<main>'
Tasks: TOP => compile:libsass => compile:libsass:x86_64-darwin17 => copy:libsass:x86_64-darwin17:2.5.1
(See full trace by running task with --trace)
```

**Comment by glebm** [comment]
> @glebm i'm a bit confused- are the precompiled binaries going to be included as part of the ruby gem?

The command build multiple gems and we should publish them all (the source one via and 4 precompiled ones). If possible, Ruby will install use the precompiled one matching the current platform by default. See for example [nokogiri](https://rubygems.org/gems/nokogiri) - you can see multiple 1.10.3 versions; this gem will look similar.

**Comment by glebm** [comment]
> @glebm hitting the following issue on osx:

@bolandrm Try running `rm -rf tmp/ pkg/ ext/libsass && git submodule update --init`.

**Comment by glebm** [comment]
Ah, I see the problem, rake compiler tries to look for `libsass.bundle` on macOS instead of `libsass.so`.

https://github.com/rake-compiler/rake-compiler/blob/9f15620e7db145d11ae2fc4ba032367903f625e3/features/support/platform_extension_helpers.rb#L5

**Comment by glebm** [comment]
@bolandrm Fix sent: #121

**Comment by betesh** [comment]
Thanks @xzyfer @bolandrm and @glebm for the quick response.

Will you be releasing a new version of sassc-ruby to rubygems.org in the near future?

**Comment by bolandrm** [comment]
Yeah, perhaps tomorrow

**Comment by betesh** [comment]
Thanks @bolandrm .  That will definitely be appreciated -- a lot easier than having to build from source myself.

**Comment by betesh** [comment]
@bolandrm Sorry to nag you; it would be really helpful if you could release this.  Thanks!

**Comment by bolandrm** [comment]
Released   `2.1.0.pre1`.  Please give it a try and let me know if it's working well.

**Comment by ahorek** [comment]
precompiled versions doesn't work for me (tested on linux and windows)

<pre>
LoadError: Could not open library '/home/ahorek/.rvm/gems/ruby-2.5.3/gems/sassc-2.1.0.pre1-x86_64-linux/lib/sassc/2.5/libsass.so': /home/ahorek/.rvm/gems/ruby-2.5.3/gems/sassc-2.1.0.pre1-x86_64-linux/lib/sassc/2.5/libsass.so: symbol _ZTVNSt7__cxx1119basic_istringstreamIcSt11char_traitsIcESaIcEEE, version GLIBCXX_3.4.21 not defined in file libstdc++.so.6 with link time reference
/home/ahorek/.rvm/gems/ruby-2.5.3/gems/ffi-1.11.1/lib/ffi/library.rb:145:in `block in ffi_lib'
/home/ahorek/.rvm/gems/ruby-2.5.3/gems/ffi-1.11.1/lib/ffi/library.rb:99:in `map'
/home/ahorek/.rvm/gems/ruby-2.5.3/gems/ffi-1.11.1/lib/ffi/library.rb:99:in `ffi_lib'
/home/ahorek/.rvm/gems/ruby-2.5.3/gems/sassc-2.1.0.pre1-x86_64-linux/lib/sassc/native.rb:15:in `<module:Native>'
/home/ahorek/.rvm/gems/ruby-2.5.3/gems/sassc-2.1.0.pre1-x86_64-linux/lib/sassc/native.rb:6:in `<module:SassC>'
/home/ahorek/.rvm/gems/ruby-2.5.3/gems/sassc-2.1.0.pre1-x86_64-linux/lib/sassc/native.rb:5:in `<top (required)>'
/home/ahorek/.rvm/gems/ruby-2.5.3/gems/sassc-2.1.0.pre1-x86_64-linux/lib/sassc.rb:31:in `require_relative'
/home/ahorek/.rvm/gems/ruby-2.5.3/gems/sassc-2.1.0.pre1-x86_64-linux/lib/sassc.rb:31:in `<top (required)>'
</pre>

**Comment by glebm** [comment]
Looks like we're not linking libc statically but we probably should.

**Comment by johnfairh** [comment]
👍 macOS 10.14.5 / Xcode 10.2 / mri 2.0.0/various/2.6.3

**Comment by betesh** [comment]
Seems to be working with jruby 9.1.17.0!  Thanks!

**Comment by bolandrm** [comment]
related:  https://github.com/sass/sassc-ruby/issues/125

@glebm will you be able to take a look at these issues?  This is a bit out of my wheelhouse.

**Comment by glebm** [comment]
@bolandm Yes, I'll take a look in a few days. I'm travelling at the moment

**Pull request merged**